server/server.go

```diff
@@ -270,7 +271,19 @@ func (s *Server) Shutdown(ctx context.Context) error {
 	// crio.service restart!!!
 	s.cleanupSandboxesOnShutdown(ctx)
 
-	return s.ContainerServer.Shutdown()
+	if err := s.ContainerServer.Shutdown(); err != nil {
+		return err
+	}
+
+	syscall.Sync()
+
+	f, err := os.Create(s.config.CleanShutdownFile)
```

Comment:
We will have to sync this file :)

Reply:
getting this atomic is challenging. If we call Create, then immediately after call Sync(), are we guarenteed to have the storage to sync before CleanShutdownFile? If not, we could believe we shutdown cleanly, but actually still have a corrupted storage. In this case, I'd rather let the Create() call be sure to happen after storage is on disk.

If you'd rather, I can add another Sync() after the create, though.

Reply:
We can do a separate fsync for this file and worst case if it fails, we end up wiping storage. The fsync for this file  is to reduce the possibility of that happening. We could still get power cable yanked after the sync and before the fsync of this file and that's okay.

Reply:
fixed

Reply:
you could just fsync the parent directory, as in the suggestion I've made above.

Reply:
I don't think it matters much to get  it atomic, in unlikely case of a crash between the file creation and the fsync for the parent directory, the file won't be found on the next reboot and the storage is wiped out